From eb2e7b5bdfd545a8278e82052060a9d7eaafc498 Mon Sep 17 00:00:00 2001
From: Zheng Zengkai <zhengzengkai@huawei.com>
Date: Fri, 9 Sep 2022 11:16:11 +0800
Subject: [PATCH] LoongArch: fix kabi change due to enum chuph_state
Patch-mainline: NO, check it manually
References: openEuler-22.03-LTS
Modified-by-SEL: No


LoongArch inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I5OHOB

--------------------------------

As commit ac1f7a067145
("irqchip: Add Loongson Extended I/O interrupt controller support")
changed the kabi of x86&arm,so add CONFIG_LOONGARCH to just change
the kabi for LoongArch.

Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/cpuhotplug.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/cpuhotplug.h b/include/linux/cpuhotplug.h
index 51f991260cce..18e017dca60a 100644
--- a/include/linux/cpuhotplug.h
+++ b/include/linux/cpuhotplug.h
@@ -106,7 +106,9 @@ enum cpuhp_state {
 	CPUHP_AP_IRQ_BCM2836_STARTING,
 	CPUHP_AP_IRQ_MIPS_GIC_STARTING,
 	CPUHP_AP_IRQ_RISCV_STARTING,
+#ifdef CONFIG_LOONGARCH
 	CPUHP_AP_IRQ_LOONGARCH_STARTING,
+#endif
 	CPUHP_AP_IRQ_SIFIVE_PLIC_STARTING,
 	CPUHP_AP_ARM_MVEBU_COHERENCY,
 	CPUHP_AP_MICROCODE_LOADER,
-- 
2.33.0

